moment_kinetics/src/z_advection.jl

```diff
@@ -93,10 +93,25 @@ function update_speed_z!(advect, upar, vth, evolve_upar, evolve_ppar, fields, vp
     if z.advection.option == "default"
         # bzed = B_z/B only used for z.advection.option == "default"
         bzed = geometry.bzed
-        ExBfac = -0.5*geometry.rhostar
+        Bmag = geometry.Bmag
+        bzeta = geometry.bzeta
+        jacobian = geometry.jacobian
+        rhostar = geometry.rhostar
+        ExBfac = -0.5*rhostar
+        geofac = z.scratch
+        cvdriftz = geometry.cvdriftz
+        gbdriftz = geometry.gbdriftz
         @inbounds begin
             @loop_r_vperp_vpa ir ivperp ivpa begin
-                @. @views advect.speed[:,ivpa,ivperp,ir] = vpa.grid[ivpa]*bzed + ExBfac*fields.Er[:,ir]
+                # vpa bzed
+                @. @views advect.speed[:,ivpa,ivperp,ir] = vpa.grid[ivpa]*bzed[:,ir]
+                # ExB drift
+                @. geofac = bzeta[:,ir]*jacobian[:,ir]/Bmag[:,ir]
+                @. @views advect.speed[:,ivpa,ivperp,ir] += ExBfac*geofac*fields.Er[:,ir]
```

Comment:
This would allocate an unnecessary intermediate array, so do in one line
```suggestion
                @. @views advect.speed[:,ivpa,ivperp,ir] += ExBfac*bzeta[:,ir]*jacobian[:,ir]/Bmag[:,ir]*fields.Er[:,ir]
```